envoy/config/filter/http/jwt_auth/v2/config.proto

```diff
@@ -0,0 +1,188 @@
+  // system wide default is applied if no duration is explicitly
+  // specified.
+  google.protobuf.Duration public_key_cache_duration = 5;
+
+  // Two fields below define where to extract the JWT from an HTTP request.
+  //
+  // If no explicit location is specified the following default
+  // locations are tried in order:
+  //
+  //     1) The Authorization header using the Bearer schema,
+  //        e.g. Authorization: Bearer <token>. (see
+  //        https://tools.ietf.org/html/rfc6750#section-2.1)
+  //
+  //     2) `access_token` query parameter (see
+  //     https://tools.ietf.org/html/rfc6750#section-2.3)
+  //
+
+  // Only one JWT token is verified per request.
+  // If multiple header or parameter names specified, the first one with the
+  // token will be used.
+  // If both jwt_headers and jwt_params are specified, jwt_headers will be
+  // checked first.
+
+  // JWT is sent in a request header. `jwt_headers` represents the
+  // header names. Their format is:
+  //    header_name: <JWT>
+  //
+  // For example, if config is
+  //   jwt_headers:
+  //      x-goog-iap-jwt-assertion
+  // The header format will be:
+  //   x-goog-iap-jwt-assertion: <JWT>.
+  //
+  repeated string jwt_headers = 6;
+
+  // JWT is sent in a query parameter. `jwt_params` represents the
+  // query parameter names.
+  //
+  // For example, if config is:
+  //   jwt_params:
+  //      jwt_token
+  // The query parameter is:  ?jwt_token=<JWT>
+  //
+  repeated string jwt_params = 8;
+
+  // This field is specific for Envoy proxy implementation.
+  // It is the cluster name in the Envoy config for the jwks_uri.
+  string jwks_uri_cluster = 7;
+}
+
+// Determines how to apply auth policies for individual requests.
+message Config {
+  // List of JWT rules to valide.
+  //
+  // If the request includes a JWT it must match one of the JWT listed
+  // here matched by the issuer. If validation is successfull the
+  // follow attributes are included in requests to the mixer:
+  //
+  //     request.auth.principal - The string of the issuer (`iss`) and
+  //     subject (`sub`) claims within a JWT concatenated with “/”
+  //     with a percent-encoded subject value
+  //
+  //     request.auth.audiences - This should reflect the audience
+  //     (`aud`) claim within matched JWT.
+  //
+  //     request.auth.presenter - The authorized presenter of the
+  //     credential. This value should reflect the optional Authorized
+  //     Presenter (`azp`) claim within a JWT
+  //
+  // If no match is found the request is rejected with HTTP status
+  // code 401.
+  //
+  // JWT validation is skipped if the user's traffic request does not
+  // include a JWT.
+  repeated JWT jwts = 2;
+
+  // This message defines a pattern to match a HTTP request.
+  // A pattern is matched only if both http_method and path_match are matched.
+  message HttpPattern {
+    // Define a HTTP method.
+    string http_method = 1;
+
+    // Defines a path match
+    oneof path_match {
```

Comment:
Can you reuse the `RouteMatch` protos in https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/route/route.proto? Seems at the code level it should be possible to refactor/share to make this easy to implement and consistent.

Reply:
Updated to use HeaderMatcher